test/gtest/ucp/test_ucp_tag_xfer.cc

```diff
@@ -244,19 +249,26 @@ void test_ucp_tag_xfer::test_xfer_generic(size_t size, bool expected, bool sync)
     dt_gen_start_count  = 0;
     dt_gen_finish_count = 0;
 
+    if ((truncated) && (!count)) {
+        truncated = false;
```

Comment:
i meant - not assign anything to 'truncated' at all. just change the 'if' in line 260 to check the value of 'count' in addition to 'truncated'.